enumerable_methods.rb

```diff
@@ -0,0 +1,202 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0...length).each do |i|
+        yield self[i]
+      end
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
```

Comment:
#### REQUIRED

There is a :bug: with `my_each_with_index` method. Please, let's fix this.

```ruby
irb(main):004:0> [1,2,3].each_with_index{|i,j| p [i, j]}
[1, 0]
[2, 1]
[3, 2]
=> [1, 2, 3]
irb(main):005:0> [1,2,3].my_each_with_index{|i,j| p [i, j]}
[1, 0]
[1, 0]
[1, 0]
=> 0...3
```

Reply:
Thank you. Fixed this also.